### ISSUE TITLE
Cross-tokenizer acceptance is a lower bound: a merged prompt/generation boundary token costs exactly 1/n_gen, always downward

### ISSUE THREAD

**Issue by MakazhanAlpamys** [MakazhanAlpamys/Soup#462]
Split out of the #417 review (cross-tokenizer drafts) so it is not lost when that PR merges. Found while reviewing, measured with a purpose-built fixture rather than left hypothetical.

## What happens

On the cross-tokenizer path, `measure_acceptance` aligns the draft's text against the target's by character span. When the draft tokenizer merges the **last prompt character with the first generated one**, the token straddling that boundary drops out of the proposals and is scored as a miss — even for a draft whose logits reproduce the target text *exactly*.

A perfect draft, varying only the boundary merge:

```
control: clean boundary          accepted=2 total=2  alpha=1.000
merged boundary (b + c -> 'bc')  accepted=1 total=2  alpha=0.500
```

## Severity: exactly one target token, and always downward

The loss is `-1/n_gen`:

| n_gen | clean | merged | loss |
|---|---|---|---|
| 2 | 1.000 | 0.500 | -0.500 |
| 8 | 1.000 | 0.875 | -0.125 |
| 24 | 1.000 | 0.958 | -0.042 |

At the default `--max-new-tokens 64` that is about **-1.6%**.

**It can only under-report alpha.** A cross-tokenizer draft can therefore read slightly worse than it is — a false "this draft is not good enough" — but never a false green. That is the safe direction, which is why this was a nit in the review rather than a blocker, and why it is a follow-up rather than a fix in #417.

It does interact with `soup draft measure --min-acceptance`, which exits 2 as a CI gate: a draft sitting within ~1.6% above the threshold could be failed by the boundary token alone.

## Acceptance

- [ ] A docstring note on the cross-tokenizer branch of `measure_acceptance` stating that alpha is a **lower bound**, short by up to one boundary token (`1/n_gen`), and that the bias is always downward.
- [ ] The fixture above as a regression test: a perfect draft scoring `1.000` on a clean boundary and `1 - 1/n_gen` on a merged one — so the size of the bias is pinned and a future change that makes it *worse* (or flips it upward, which would be the unsafe direction) fails.
- [ ] Optionally: mention the bound in `docs/serving-and-export.md` next to the acceptance-rate explanation, where `--min-acceptance` is described.

Not blocking #417 — that PR's own scope is the cross-tokenizer measurement itself, and this is a known, bounded, safe-direction bias in it.

**Comment by MakazhanAlpamys** [comment]
**Claimed by @kok-o** — recording it here so the issue page shows it, since the work arrived as a PR (#464) before a claim.

One scheduling note that belongs on this issue rather than only in that PR: the code this asks you to annotate lives on #417's branch and is **not on `main` yet**, so #464 currently carries that PR's unmerged commits alongside yours. The plan is to wait for #417 to merge, then rebase — at which point #464 becomes exactly the one commit it should be.

My wording here ("Not blocking #417") was the source of the confusion: I meant this issue does not gate that PR, not that the work could start before it lands. Corrected for the next reader.

**Comment by MakazhanAlpamys** [comment]
Fixed by @kok-o in #464, merged as `41c7645`.

The caveat is now documented on the cross-tokenizer branch of `measure_acceptance`, and the bias is pinned at exactly `1/n_gen` by a regression test.

**The review is worth recording, because the first two attempts were both wrong in instructive ways:**

1. The fixture fed a merged token **into** the kernel. But `count_accepted_spans` takes a `d_text == t_text` early return there — correctly, since a draft that reproduces the target text exactly *should* score 1.0 regardless of tokenization. The real bias is a **shorter proposal list**: `measure_acceptance`'s `[draft_prompt_len - 1 : n - 1]` slice never proposes the straddling token.
2. It used a 2-token target, where "dropped exactly one token" and "dropped half the tokens" are **both 0.50**. The issue's whole claim is `1/n_gen`, so a 2-token fixture pins "50%" just as well — at 4 tokens they are 0.75 and 0.50 and the claim is actually encoded.

Verified on the merged head: the unsafe direction — a kernel that **over**-reports acceptance, making a draft look better than it is — fails the new test by name.

One thing deliberately left unpinned: deleting the docstring note breaks nothing. A test asserting on prose is often worse than the prose.

Try:

```
soup draft measure --target ./tuned --draft HuggingFaceTB/SmolLM2-135M --prompts p.jsonl
```
(alpha on a cross-tokenizer pair is a lower bound, short by up to one boundary token)
